src/main/java/org/opensearch/knn/indices/ModelGraveyard.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.indices;
+
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.Version;
+import org.opensearch.cluster.Diff;
+import org.opensearch.cluster.NamedDiff;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import com.google.common.collect.Sets;
+
+/**
+ * This class implements Metadata.Custom Interface to store a set of modelIds in the cluster metadata.
+ * The modelIds of the models that are under deletion are added to this set and later removed from this set after deletion.
+ * Also, this class implements the methods to perform operations on this set (like add, remove, contains)
+ */
+@Log4j2
+public class ModelGraveyard implements Metadata.Custom {
+    public static final String TYPE = "opensearch-knn-blocked-models";
+    private final Set<String> modelGraveyard;
```

Comment:
I would rename this to modelIds. The class is named ModelGraveyard so having a member called modelGraveyard is confusing to me.

For instance, look at:
```
modelGraveyard.getModelGraveyard()
```

This seems a little confusing - we already have the modelGraveyard, so what are we getting?

Reply:
Agreed. I have updated it